server/test/todo.test.js

```diff
@@ -0,0 +1,106 @@
+'use strict';
+
+var lt = require('loopback-testing');
+var assert = require('assert');
+var app = require('../server.js'); //path to app.js or server.js
+
+describe('Todo', function() {
+  lt.beforeEach.withApp(app);
+  lt.beforeEach.givenModel('Todo');
+
+  // New todo
+  lt.describe.whenCalledRemotely('POST', '/api/Todos', {}, function() {
+    lt.it.shouldBeAllowed();
+    it('should have statusCode 200', function() {
+      assert.equal(this.res.statusCode, 200);
+    });
+
+    it('should respond with a new todo', function() {
+      assert(this.res.body.id.match(/t-\d+/));
+    });
+  });
+
+  // Get todo stats
+  lt.describe.whenCalledRemotely('POST', '/api/Todos/stats', function() {
+    it('should respond with total, remaining and 1 completed', function() {
+      lt.it.shouldBeAllowed();
+      assert.equal(typeof this.res.body, 'object');
+      assert.equal(this.res.body.stats.remaining, 4);
```

Comment:
Where are the 4 todo items coming from? The test seems to create two of them only - one in `lt.beforeEach.givenModel('Todo')` and second by `lt.describe.whenCalledRemotely('POST', '/api/Todos', {}`